File: app/src/main/java/com/reitler/got/model/statistics/MatchStatistic.java
package com.reitler.got.model.statistics;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class MatchStatistic {

    private List<Integer> scores = new ArrayList<>();

    public void add(int score){
        this.scores.add(score);
    }

    public int getScoreSum(){
        return scores.stream().collect(Collectors.summingInt(Integer::intValue));
    }

    public int getPlayerCount(){
        return scores.size();
    }

    public double getAvgPlayerScore(){
        return ((double) getScoreSum()) / getPlayerCount();
    }
}

File: app/src/main/java/com/reitler/got/model/statistics/MatchStatistic.kt
package com.reitler.got.model.statistics

import java.util.*
import java.util.stream.Collectors

class MatchStatistic {
    private val scores: MutableList<Int> = ArrayList()

    fun add(score: Int) {
        scores.add(score)
    }

    val scoreSum: Int
        get() = scores.sum()
    val playerCount: Int
        get() = scores.size
    val avgPlayerScore: Double
        get() = scoreSum.toDouble() / playerCount
    val maxScore: Int
        get() = scores.maxOf { it }
}